packages/react-native-editor/__device-tests__/helpers/utils.js

```diff
@@ -454,6 +449,35 @@ const toggleOrientation = async ( driver ) => {
 	}
 };
 
+const isEditorVisible = async ( driver ) => {
+	const postTitleLocator = isAndroid()
+		? `//android.widget.EditText[contains(@content-desc, "Post title")]`
+		: `(//XCUIElementTypeScrollView/XCUIElementTypeOther/XCUIElementTypeOther[contains(@name, "Post title")])`;
+
+	await waitForVisible( driver, postTitleLocator );
+};
+
+const waitForVisible = async ( driver, elementLocator, iteration = 0 ) => {
+	const maxIteration = 25;
+	const timeout = 1000;
+
+	if ( iteration >= maxIteration ) {
+		throw new Error(
+			`"${ elementLocator }" is still not visible after ${ iteration } retries!`
+		);
+	} else if ( iteration !== 0 ) {
+		// wait before trying to locate element again
+		await driver.sleep( timeout );
+	}
+
+	const locator = await driver.elementsByXPath( elementLocator );
+	if ( locator.length !== 1 ) {
+		// if locator is not visible, try again
+		return waitForVisible( driver, elementLocator, iteration + 1 );
+	}
+	return locator[ 0 ];
```

Comment:
Thanks for making this change @fluiddot! I can now see why it wasn't working when I tried earlier, I sent everything instead of just the first item in the array.